developers/academy/js/starter_multimodal_data/_snippets/101_connect.mts

```diff
@@ -0,0 +1,171 @@
+// DockerInstantiation // WCDInstantiation // WCDAPIKeyInstantiation // DockerInstantiation // DockerAPIKeyInstantiation // TryFinallyCloseDemo
+import weaviate, { WeaviateClient } from "weaviate-client";
+let client: WeaviateClient;
+// END DockerInstantiation // END WCDInstantiation // END WCDAPIKeyInstantiation // END DockerInstantiation // END DockerAPIKeyInstantiation // END TryFinallyCloseDemo
+
+
+// WCDInstantiation
+
+client = await weaviate.connectToWeaviateCloud(
+  process.env.WCD_URL as string, {
+  authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
+})
+
+// END WCDInstantiation
+
+client.close()
+
+// WCDAPIKeyInstantiation
+
+client = await weaviate.connectToWeaviateCloud(
+  process.env.WCD_URL as string, {
+  authCredentials: new weaviate.ApiKey(process.env.WCD_API_KEY as string),
+  headers: {
+    'X-Cohere-Api-Key': process.env.COHERE_APIKEY as string,  // Replace with your inference API key
+  }
+})
+// END WCDAPIKeyInstantiation
+
+client.close()
+
+// DockerInstantiation
+
+client = await weaviate.connectToLocal()
+// END DockerInstantiation
+
+client.close()
+
+// DockerAPIKeyInstantiation
+
+client = await weaviate.connectToLocal({
+  host: '...',
+  headers: {
+    'X-Cohere-Api-Key': process.env.COHERE_APIKEY as string,  // Replace with your inference API key
+  }
+})
+// END DockerAPIKeyInstantiation
+
+// PollLiveness
+if (await client.isLive()) {
+  // This will raise an exception if the client is not live
```

Comment:
This looks like just a true/false control loop - will it raise an exception?

Reply:
no, we raise the exception outside the if statement. I'll change wording